ckanext/harvest/queue.py

```diff
@@ -202,20 +202,44 @@ def consume(self, queue):
             key, body = self.redis.blpop(self.routing_key)
             self.redis.set(self.persistance_key(body),
                            str(datetime.datetime.now()))
+            print('SET IT!')
             yield (FakeMethod(body), self, body)
 
     def persistance_key(self, message):
-        # Persistance keys are constructed with
-        # {site-id}:{message-key}:{object-id}, eg:
-        # default:harvest_job_id:804f114a-8f68-4e7c-b124-3eb00f66202e
+        # If you change this, make sure to update the script in `queue_purge`
         message = json.loads(message)
         return self.routing_key + ':' + message[self.message_key]
 
     def basic_ack(self, message):
         self.redis.delete(self.persistance_key(message))
 
-    def queue_purge(self, queue):
-        self.redis.flushdb()
+    def queue_purge(self, queue=None):
+        '''
+        Purge the consumer's queue.
+
+        The ``queue`` parameter exists only for compatibility and is
+        ignored.
```

Comment:
I don't think so. As far as I can tell, `RedisConsumer` tries to look like an AMQP channel, whose `queue_purge` method *does* take the parameter. To be honest, the separation between the AMQP and Redis back-ends isn't that clear in the code so I tried to make as little changes as necessary.